src/Guide/Views.hs

```diff
@@ -838,56 +839,90 @@ renderAdminLinks globalState = do
+      --   case d of
+      --     Left err -> putStrLn err
+      --     Right ps -> print $ toProperDate $ timestamp $ closest $ archivedSnapshots ps
+      --   pure respArch
+      --
+      -- div_ $ toHtml $ toStrict archiveAnswer
+      -- div_ $ button "To Archive" [] (JS.saveToArchiveOrg [JS.toJS $ T.pack "example.com"])
       table_ [class_ "sortable"] $ do
         thead_ $ tr_ $ do
           th_ [class_ "sorttable_nosort"] "Category"
           th_ [class_ "sorttable_nosort"] "Link"
           th_ "Status"
-        tbody_ $ do
-          for_ broken $ \(location, lnk, reason) -> do
+          th_ "Saved page"
+        tbody_ $
+          for_ broken $ \(location, lnk, reason, (_, archUrl)) ->
             tr_ $ do
               td_ location
-              td_ lnk
+              td_ $ a_ [href_ lnk] (toHtml lnk)
               td_ $ toHtml reason
+              td_ archUrl
       h2_ "Unparseable Links"
       table_ [class_ "sortable"] $ do
           thead_ $ tr_ $ do
             th_ [class_ "sorttable_nosort"] "Category"
             th_ [class_ "sorttable_nosort"] "Link"
-          tbody_ $ do
-            for_ unparseable $ \(cat, l) -> do
+          tbody_ $
+            for_ unparseable $ \(location, lnk) ->
               tr_ $ do
-                td_ cat
-                td_ l
+                td_ location
+                td_ $ a_ [href_ lnk] (toHtml lnk)
       h2_ "OK Links"
       table_ [class_ "sortable"] $ do
           thead_ $ tr_ $ do
             th_ [class_ "sorttable_nosort"] "Category"
             th_ [class_ "sorttable_nosort"] "Link"
-          tbody_ $ do
-            for_ ok $ \(cat, l) -> do
+            th_ [class_ "sorttable_nosort"] "Latest archieve date"
+            th_  "Save to Archive"
+          tbody_ $
+            for_ ok $ \(location, lnk, (dt, _)) ->
               tr_ $ do
-                td_ cat
-                td_ l
+                td_ location
+                td_ $ a_ [href_ lnk] (toHtml lnk)
+                td_ dt
+                td_ $ button "To Archive" [] (JS.saveToArchiveOrg [JS.toJS lnk])
  where
   handleHttpException :: HttpException -> LinkStatus
   handleHttpException (HttpExceptionRequest _ x) = Broken $ show x
   handleHttpException (InvalidUrlException  _ x) = Broken x
 
-  sortLinks :: [(a, b, LinkStatus)] -> ([(a, b)], [(a, b)], [(a, b, String)])
+  handleHttpExceptionDecode :: HttpException -> IO (String, String)
+  handleHttpExceptionDecode _ = pure ("error", "error")
+
+  sortLinks :: [(a, b, LinkStatus, c)] -> ([(a, b, c)], [(a, b)], [(a, b, String, c)])
   sortLinks = foldr sortLink ([], [], [])
 
-  sortLink (a, b, OK)          = (\(x, y, z) -> ((a, b):x, y, z))
-  sortLink (a, b, Unparseable) = (\(x, y, z) -> (x, (a, b):y, z))
-  sortLink (a, b, Broken text') = (\(x, y, z) -> (x, y, (a, b, text'):z))
+  sortLink (a, b, OK, c)           = \ (x, y, z) -> ((a, b, c) : x, y, z)
```

Comment:
I think that once we are doing four-tuples, it's better to create a proper data type in this module and use it instead.